files/nrpe/check_object.py

```diff
@@ -144,6 +170,56 @@ def execute(self):
     except:
       raise
 
+class SwiftPublicAvailability():
+  '''
+  Check Swift API call length by listing contents of a public bucket
+  '''
+  options = dict()
+
+  def __init__(self, options):
+     self.options = options
+     if options.s3_host is None:
+       raise CredentialsMissingException(key='s3_host')
+     if options.s3_bucket_url is None:
+       raise CredentialsMissingException(key='s3_bucket_url')
+     self.creds = OSCredentials(options)
+
+  def list_public_swift_objects(self):
+    """ read a public swift URL with urllib
+    This does not create a public bucket if one does not exist.
+    This fails if the bucket does not exist or if it's private.
+    """
+    msgs = []
+    auth = identity.v3.Password(**self.creds.provide_keystone_v3())
+    session_ = session.Session(auth=auth)
+    keystone = keystoneclientv3.Client(session=session_)
+
+    project_id = keystone.projects.client.get_project_id()
+
+    if project_id == None:
+      raise ProjectNotAvailableException(msgs=self.options.tenant)
+
+    s3_url = "{}/AUTH_{}/{}-{}".format(self.options.s3_host,project_id,self.options.tenant,self.options.s3_bucket_url)
```

Comment:
As discussed in call with Chris,

1) previous form was AUTH_PROJECT_ID/PROJECT_NAME_yadda_yadda
2) to simplify the implementation we can drop the PROJECT_NAME.

Reply:
Yes.